### ISSUE TITLE
CatboostError: Model must not be trained for multiclassification.

### ISSUE THREAD

**Issue by saran46** [catboost/catboost#474]
Problem: Model must not be trained for multiclassification.
catboost version: 0.10.1
Operating System: macos 10.13.6
CPU: 2015 i7 2.2Ghz
GPU: None

I am learning catboost by following along the the tutorials from Github
https://github.com/catboost/catboost/blob/master/catboost/tutorials/advanced_tutorials/shap_values_tutorial.ipynb
When i calculate shap values for multiclass using the model with **MultiClass** loss function i run into following error

CatboostError: catboost/libs/fstr/shap_values.cpp:560: Model must not be trained for multiclassification.

I am running
catboost 0.10.1
shap 0.24.0

```
from catboost.datasets import *
import shap
shap.initjs()

train_df, test_df = catboost.datasets.amazon()
y = train_df.ACTION
X = train_df.drop('ACTION', axis=1)
cat_features = list(range(0, X.shape[1]))

model = CatBoostClassifier(loss_function = 'MultiClass', iterations=300, learning_rate=0.1, random_seed=123)
model.fit(X, y, cat_features=cat_features, verbose=False, plot=False)

explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(Pool(X, y, cat_features=cat_features))
```

```
---------------------------------------------------------------------------
CatboostError                             Traceback (most recent call last)
<ipython-input-29-9ad8f66ea226> in <module>()
      1 explainer = shap.TreeExplainer(model)
----> 2 shap_values = explainer.shap_values(Pool(X, y, cat_features=cat_features))

~/anaconda3/lib/python3.6/site-packages/shap/explainers/tree.py in shap_values(self, X, tree_limit, approximate)
    161             if type(X) != catboost.Pool:
    162                 X = catboost.Pool(X)
--> 163             phi = self.trees.get_feature_importance(data=X, fstr_type='ShapValues')
    164 
    165         # note we pull off the last column and keep it as our expected_value

~/anaconda3/lib/python3.6/site-packages/catboost/core.py in get_feature_importance(self, data, fstr_type, prettified, thread_count, verbose)
   1507 
   1508         with log_fixup():
-> 1509             fstr, feature_names = self._calc_fstr(fstr_type, data, thread_count, verbose)
   1510         if fstr_type == EFstrType.FeatureImportance:
   1511             feature_importances = [value[0] for value in fstr]

~/anaconda3/lib/python3.6/site-packages/catboost/core.py in _calc_fstr(self, fstr_type, pool, thread_count, verbose)
    884     def _calc_fstr(self, fstr_type, pool, thread_count, verbose):
    885         """returns (fstr_values, feature_ids)."""
--> 886         return self._object._calc_fstr(fstr_type.name, pool, thread_count, verbose)
    887 
    888     def _calc_ostr(self, train_pool, test_pool, top_size, ostr_type, update_method, importance_values_sign, thread_count):

_catboost.pyx in _catboost._CatBoost._calc_fstr()

_catboost.pyx in _catboost._CatBoost._calc_fstr()

CatboostError: catboost/libs/fstr/shap_values.cpp:560: Model must not be trained for multiclassification.
```

Kindly let me know, how to fix this issue.

Thank You.

**Comment by annaveronika** [comment]
closing because of inactivity

**Comment by khrisanfov** [comment]
@annaveronika Hello! I have the same issue on CatBoost 0.10.3. You can reproduce the issue by following the tutorial steps https://github.com/catboost/tutorials/blob/master/advanced_tutorials/shap_values_tutorial.ipynb

Code:
```
print(catboost.version.VERSION)
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(Pool(X, y, cat_features=cat_features))
```
Output:
```
0.10.3
---------------------------------------------------------------------------
CatboostError                             Traceback (most recent call last)
<ipython-input-22-79fd3868a014> in <module>
      1 print(catboost.version.VERSION)
      2 explainer = shap.TreeExplainer(model)
----> 3 shap_values = explainer.shap_values(Pool(X, y, cat_features=cat_features))

/usr/local/lib/python3.6/dist-packages/shap/explainers/tree.py in shap_values(self, X, tree_limit, approximate)
    150             if type(X) != catboost.Pool:
    151                 X = catboost.Pool(X)
--> 152             phi = self.trees.get_feature_importance(data=X, fstr_type='ShapValues')
    153 
    154         # note we pull off the last column and keep it as our expected_value

/usr/local/lib/python3.6/dist-packages/catboost/core.py in get_feature_importance(self, data, fstr_type, prettified, thread_count, verbose)
   1536 
   1537         with log_fixup():
-> 1538             fstr, feature_names = self._calc_fstr(fstr_type, data, thread_count, verbose)
   1539         if fstr_type == EFstrType.FeatureImportance:
   1540             feature_importances = [value[0] for value in fstr]

/usr/local/lib/python3.6/dist-packages/catboost/core.py in _calc_fstr(self, fstr_type, pool, thread_count, verbose)
    894     def _calc_fstr(self, fstr_type, pool, thread_count, verbose):
    895         """returns (fstr_values, feature_ids)."""
--> 896         return self._object._calc_fstr(fstr_type.name, pool, thread_count, verbose)
    897 
    898     def _calc_ostr(self, train_pool, test_pool, top_size, ostr_type, update_method, importance_values_sign, thread_count):

_catboost.pyx in _catboost._CatBoost._calc_fstr()

_catboost.pyx in _catboost._CatBoost._calc_fstr()

CatboostError: catboost/libs/fstr/shap_values.cpp:565: Model must not be trained for multiclassification.
```

**Comment by khrisanfov** [comment]
@nikitxskv @annaveronika could anyone reopen this issue?

**Comment by dbakshee** [comment]
Hi, all
The issue should be fixed by commit 145ad6b22ee390d6d255448703383aa8c120cc0d

The reason was that bytes did not correctly compare to string, particularly [here](https://github.com/catboost/catboost/blob/master/catboost/python-package/catboost/_catboost.pyx#L1623)

**Comment by annaveronika** [comment]
ok, than I'm closing again. You can build from source now: https://tech.yandex.com/catboost/doc/dg/installation/python-installation-method-build-from-source-linux-macos-docpage/

Or wait for our next pypi release, which will happen this week.
